build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -1416,6 +1416,108 @@ export const handlers: Handler[] = [
 				}
 			});
 
+			return result;
+		},
+	},
+	{
+		// Verify that non-private packages are configured correctly for API-Extractor by checking for the following:
+		// - dev dependency on `@microsoft/api-extractor`
+		// - `build:docs` script
+		// - `check:release-tags` script
+		name: "api-extractor-public-packages",
+		match,
+		handler: async (file) => {
+			let json: PackageJson;
+			try {
+				json = JSON.parse(readFile(file));
+			} catch (err) {
+				return "Error parsing JSON file: " + file;
+			}
+
+			const packageName = json.name;
+
+			if (json.private || packageName.startsWith("@fluid-private")) {
+				// If the package is private, we have nothing to validate.
+				return;
+			}
+
+			const errors: string[] = [];
+
+			// Ensure the package has a dev dependency on `@microsoft/api-extractor`
+			const devDependencies = Object.keys(json.devDependencies ?? {});
+			if (!devDependencies.includes("@microsoft/api-extractor")) {
+				errors.push(`Missing dev dependency: "@microsoft/api-extractor"`);
+			}
+
+			/**
+			 * Creates an error string noting the package.json is missing a script of the provided name.
+			 * @param missingScriptName - The name of the missing script.
+			 */
+			function createMissingScriptErrorString(missingScriptName): string {
+				return `Missing script: "${missingScriptName}"`;
+			}
+
+			const scripts = Object.keys(json.scripts ?? {});
+
+			// Ensure `build:docs` script exists
+			if (!scripts.includes("build:docs")) {
+				errors.push(createMissingScriptErrorString("build:docs"));
+			}
+
+			// Ensure `check:release-tags` script exists
+			if (!scripts.includes("check:release-tags")) {
+				errors.push(createMissingScriptErrorString("check:release-tags"));
+			}
+
+			// Ensure `ci:build:docs` script exists
+			if (!scripts.includes("ci:build:docs")) {
+				errors.push(createMissingScriptErrorString("ci:build:docs"));
+			}
+
+			// TODO: any other scripts to check? `apis`? `api-extractor:*`?
```

Comment:
@tylerbutler Curious which scripts you think we should check for here? I know some of the tsc-multi work is still in flux, so I don't want to over-validate here.

Reply:
Looks like this policy is primarily checking for required scripts, so consider making the list of required scripts a config setting. That way we don't have to update the policy code if that list changes. We have a similar policy that checks that if a script uses a command, the dependency with that command is also included. The mapping is configurable which makes it much easier to introduce new requirements.

Reply:
I've refactored the code to use a configuration object for "public package requirements", which include required dev dependencies and required scripts (for now). Let me know what you think of this pattern.